BugFix: prevent addCustomLine from crashing without custom line points

SlySven opened the pull request:
Providing an empty table as the second argument (i.e. without sets of sub-tables as x, y, z coordinates) was causing a fatal crash.

This will close issue #5272.

Also:
* fix up some other error messages in the addCustomLine function that referred to the wrong argument index - or didn't mention which one was involved at all.
* silence some warnings in the same function caused by casting floating point numbers to integers or down-casting 64-bit `long long int`s to (32-bit) `int`s.
* fix handling for the fifth argument NOT being a three-color component (RGB) table - so that it throws a `lua_error(...)` if a table is not provided - previously it was ignoring this defect.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: prevent crash with no custom line points in addCustomLine

Providing an empty table as the second argument (i.e. without sets of
sub-tables as x, y, z coordinates) was causing a fatal crash.

This will close issue #5272.

Also:
* fix up some other error messages in the addCustomLine function that
  referred to the wrong argument index - or didn't mention which one was
  involved at all.
* silence some warnings in the same function caused by casting floating
  point numbers to integers or down-casting 64-bit `long long int`s to
  (32-bit) `int`s.
* fix handling for the fifth argument NOT being a three-color component
  (RGB) table - so that it throws a `lua_error(...)` if a table is not
  provided - previously it was ignoring this defect.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

keneanung commented:
/create links

keneanung reviewed (approved):
lgtm

vadi2 commented on src/TLuaInterpreter.cpp:
```diff
@@ -7844,6 +7844,16 @@ int TLuaInterpreter::addCustomLine(lua_State* L)
+            // one but there is nothing in any of the QLists and things will
+            // still blow up as per Issue #5272 - so also check for at least one
+            // x-coordinate value:
+            return warnArgumentValue(L, __func__, "no coordinate of points for the custom line given in table as second argument; at least one set of coordinates as a sub-table is required");
```
```suggestion
            return warnArgumentValue(L, __func__, "missing coordinates to create the line to");
```

SlySven replied:
I do not think that suggestion for the message gives *enough* detail, how about:
```suggestion
            return warnArgumentValue(L, __func__, "no custom line points given; at least one set (in an inner table) of coordinates as the second argument is needed");
```
which is a *bit* shorter?

vadi2 replied:
The opinion is with the first proposal... :grinning:

SlySven replied:
Sod it, I can't be arsed to fight this - though it does go against my general feelings that the error messages should give some idea about what is needed to solve the problem without having to go and read up on it on-line...

☹️

SlySven committed:
Update src/TLuaInterpreter.cpp

Popular opinion seems to say the the error message was too much information...

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>
